task-list/api/main.py

```diff
@@ -1,69 +1,231 @@
-from fastapi import FastAPI, HTTPException, Depends, status
-from sqlalchemy.orm import Session
-from sqlalchemy import or_
+import csv
 import datetime
+import json
+import logging
+
+from io import StringIO
+from typing import List
+
 from faker import Faker
-import schemas, models
+from fastapi import FastAPI, Depends, status, UploadFile, HTTPException
+from fastapi.param_functions import File
+from sqlalchemy import or_
+from sqlalchemy.orm import Session
 
-from db import get_db, engine, Base
+from api.helpers.log import log
+from api.helpers.procees_data import process_csv
+from api.models import TaskEntity
+from api.schemas import TaskSchema, BulkDeleteData
+from api.service.task_service import TaskService
+from api.db import get_db, engine, Base, SessionLocal
 
 # Initialize FastAPI app
 app = FastAPI()
 
+
 # Create the database tables
-Base.metadata.create_all(bind=engine)
+# Base.metadata.create_all(bind=engine)
+
 
 @app.get("/")
 def root():
     return "Task API is running"
 
+
 # Get all tasks
+# @app.get("/tasks")
+# def get_tasks(db: Session = Depends(get_db), search: str = ''):
+#     tasks = db.query(TaskEntity).filter(or_(
+#         TaskEntity.title.contains(search),
+#         TaskEntity.description.contains(search))).all()
+#     return {"status": "success", "task": tasks}
+
+
 @app.get("/tasks")
-def get_tasks(db: Session = Depends(get_db), search: str = ''):
-    tasks = db.query(models.TaskEntity).filter(or_(
-        models.TaskEntity.title.contains(search),
-        models.TaskEntity.description.contains(search))).all()
-    return {"status": "success", "task": tasks}
+def get_tasks():
+    log.info("Starting session")
+    session = SessionLocal()
+    service = TaskService(session)
```

Comment:
I think you could clean this up using FastAPI Depends. `ts = Depends(TaskService)` and make your services resolve a session or just pass in the session. Example here https://bitestreams.com/blog/fastapi_sqlalchemy/ (look for '5. One session per request'). section.